tahrir_api/scripts/populatedb.py

```diff
@@ -0,0 +1,217 @@
+
+import datetime
+import os
+import random
+import sys
+import transaction
+
+from sqlalchemy import engine_from_config
+from sqlalchemy.orm import scoped_session, sessionmaker
+from paste.deploy import appconfig
+from zope.sqlalchemy import ZopeTransactionExtension
+
+from ..model import (
+    Issuer,
+    Badge,
+    Person,
+    Team,
+    Milestone,
+    Series,
+    Assertion,
+    DeclarativeBase,
+)
+from ..utils import convert_name_to_id
+
+
+DBSession = scoped_session(sessionmaker(extension=ZopeTransactionExtension(),
+                                        expire_on_commit=False))
+
+
+def usage(argv):
+    cmd = os.path.basename(argv[0])
+    print('usage: %s <config_uri>\n'
+          '(example: "%s development.ini")' % (cmd, cmd))
+    sys.exit(1)
+
+
+def _getpathsec(config_uri, name):
+    if '#' in config_uri:
+        path, section = config_uri.split('#', 1)
+    else:
+        path, section = config_uri, 'main'
+    if name:
+        section = name
+    return path, section
+
+
+def main(argv=sys.argv):
+    if len(argv) != 2:
+        usage(argv)
+
+    config_uri = argv[1]
+    path, section = _getpathsec(config_uri, "pyramid")
+    config_name = 'config:%s' % path
+    here_dir = os.getcwd()
+
+    global_conf = None
+    if 'OPENSHIFT_APP_NAME' in os.environ:
+        if 'OPENSHIFT_MYSQL_DB_URL' in os.environ:
+            template = '{OPENSHIFT_MYSQL_DB_URL}{OPENSHIFT_APP_NAME}'
+        elif 'OPENSHIFT_POSTGRESQL_DB_URL' in os.environ:
+            template = '{OPENSHIFT_POSTGRESQL_DB_URL}{OPENSHIFT_APP_NAME}'
+
+        global_conf = {
+            'sqlalchemy.url': template.format(**os.environ)
+        }
+
+    settings = appconfig(config_name, name=section, relative_to=here_dir,
+                         global_conf=global_conf)
+
+    engine = engine_from_config(settings, 'sqlalchemy.')
+    DBSession.configure(bind=engine)
+    DeclarativeBase.metadata.create_all(engine)
```

Comment:
I would expect this project to have the above boiler plate code as a function that can be imported and used by the various things that need a DB connection. Bodhi has done the pattern here in the past, and I find it difficult to manage so I've been working to remove blocks of code like this and have them use a common helper function. I recommend that you consider doing that here, even if this script is the only thing to use it at first.